src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -1010,77 +850,356 @@ private Assembly LoadAssemblyHelper(string assemblyName)
             // or file corrupted.
             catch (System.IO.FileLoadException) { }
 
-            if (loadedAssembly != null)
-                return loadedAssembly;
+            return loadedAssembly;
+        }
 
-            return null;
+        private IEnumerable<PortableExecutableReference> GetPortableExecutableReferences()
+        {
+            if (ReferencedAssemblies.Length > 0)
+            {
+                var tempReferences = new List<PortableExecutableReference>(s_autoReferencedAssemblies.Value);
+                foreach (string assembly in ReferencedAssemblies)
+                {
+                    if (string.IsNullOrWhiteSpace(assembly)) { continue; }
+                    string resolvedAssemblyPath = ResolveAssemblyName(assembly, true);
+
+                    // Ignore some specified reference assemblies
+                    string fileName = PathType.GetFileName(resolvedAssemblyPath);
+                    if (s_refAssemblyNamesToIgnore.Value.Contains(fileName))
+                    {
+                        WriteVerbose(StringUtil.Format(AddTypeStrings.ReferenceAssemblyIgnored, resolvedAssemblyPath));
+                        continue;
+                    }
+                    tempReferences.Add(MetadataReference.CreateFromFile(resolvedAssemblyPath));
+                }
+                return tempReferences;
+            }
+            else
+            {
+                return s_defaultAssemblies.Value;
+            }
         }
 
         private void WriteTypes(Assembly assembly)
         {
             WriteObject(assembly.GetTypes(), true);
         }
 
-        private void CompileSourceToAssembly(string source)
+        #endregion LoadAssembly
+
+        #region SourceCodeProcessing
+
+        private OutputKind OutputAssemblyTypeToOutputKind(OutputAssemblyType outputType)
+        {
+            switch (outputType)
+            {
+                case OutputAssemblyType.Library:
+                    return OutputKind.DynamicallyLinkedLibrary;
+                case OutputAssemblyType.ConsoleApplication:
+                    return OutputKind.ConsoleApplication;
+                case OutputAssemblyType.WindowsApplication:
+                    return OutputKind.WindowsApplication;
+                default:
+                    throw new ArgumentOutOfRangeException("outputType");
+            }
+        }
+
+        private CommandLineArguments DefaultArgsParse(IEnumerable<string> args)
```

Comment:
The method name `DefaultArgsParse` should be renamed to `ParseCompilerOption`.

Reply:
Renamed.